src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const currentState = { ...state };
+  const stateArr = [];
+
+  for (const el of actions) {
+    switch (el.type) {
+      case 'addProperties':
+        Object.assign(currentState, el.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keysRemove of el.keysToRemove) {
+          delete currentState[keysRemove];
+        };
+        break;
+
+      case 'clear':
+        for (const key in currentState) {
+          if (currentState.hasOwnProperty([key])) {
+            delete currentState[key];
+          }
```

Comment:
you don't need `hasOwnProperty` here, delete will ignore keys which doesn't exist